src/main/kotlin/com/coder/gateway/CoderGatewayConnectionProvider.kt

```diff
@@ -93,31 +109,38 @@ class CoderGatewayConnectionProvider : GatewayConnectionProvider {
             cli.login(client.token)
 
             indicator.text = "Configuring Coder CLI..."
-            cli.configSsh(client.agentNames(workspaces))
-
-            // TODO: Ask for these if missing.  Maybe we can reuse the second
-            //  step of the wizard?  Could also be nice if we automatically used
-            //  the last IDE.
-            if (parameters[IDE_PRODUCT_CODE].isNullOrBlank()) {
-                throw IllegalArgumentException("Query parameter \"$IDE_PRODUCT_CODE\" is missing")
-            }
-            if (parameters[IDE_BUILD_NUMBER].isNullOrBlank()) {
-                throw IllegalArgumentException("Query parameter \"$IDE_BUILD_NUMBER\" is missing")
-            }
-            if (parameters[IDE_PATH_ON_HOST].isNullOrBlank() && parameters[IDE_DOWNLOAD_LINK].isNullOrBlank()) {
-                throw IllegalArgumentException("One of \"$IDE_PATH_ON_HOST\" or \"$IDE_DOWNLOAD_LINK\" is required")
-            }
-
-            // Check that both the domain and the redirected domain are
-            // allowlisted.  If not, check with the user whether to proceed.
-            verifyDownloadLink(parameters)
-
-            // TODO: Ask for the project path if missing and validate the path.
-            val folder = parameters[FOLDER] ?: throw IllegalArgumentException("Query parameter \"$FOLDER\" is missing")
-
-            parameters
-                .withWorkspaceHostname(CoderCLIManager.getHostName(deploymentURL.toURL(), agent.name))
-                .withProjectPath(folder)
+            cli.configSsh(client.agentNames(workspaces).toSet().toList())
+
+
+            val openDialog = if (parameters[IDE_PRODUCT_CODE].isNullOrBlank())
+                true
+            else if (parameters[IDE_BUILD_NUMBER].isNullOrBlank())
+                true
+            else if (parameters[IDE_PATH_ON_HOST].isNullOrBlank() && parameters[IDE_DOWNLOAD_LINK].isNullOrBlank())
+                true
+            else if (parameters[FOLDER].isNullOrBlank())
+                true
+            else
+                false
```

Comment:
There is probably a more idiomatic way to define `openDialog` ... 🤷

Reply:
@coryb definitely a nit, but you could collapse this all into one elongated assignment statement, rather than a series of `if` checks that all result in `true`:

```
 val openDialog = parameters[IDE_PRODUCT_CODE].isNullOrBlank() ||
      parameters[IDE_BUILD_NUMBER].isNullOrBlank() ||
      (parameters[IDE_PATH_ON_HOST].isNullOrBlank() && parameters[IDE_DOWNLOAD_LINK].isNullOrBlank()) ||
      parameters[FOLDER].isNullOrBlank()
```

Reply:
updated, I feel like I am missing some fancy kotlin syntax, but your suggestion is probably the most straight-forward.

Reply:
fixed.